### ISSUE TITLE
Broken components in 0.15.0

### ISSUE THREAD

**Issue by yanickrochon** [react-materialize/react-materialize#108]
My forms no longer display values 

```
<Input s={6} m={4} l={3} ref="qty" type="number" name="qty" label="Quantity"
   validate={true}
   value={ this.state.model.qty || model.qty || '' }
   onChange={ this.updateModel.bind(this, 'qty') }
   min="0"
/>
```

I see in the code that I must specify `defaultValue` instead. This should be documented as a breaking change, explain why, or re-implement the feature.

Also, I get this error

> Warning: Failed prop type: Invalid prop `defaultValue` of type `number` supplied to `Input`, expected `string`

because my field is a `type="number"`.

**Comment by yanickrochon** [comment]
Also, the `name` attribute is not set on the field (since it is no longer passed down), and breaks `jquery-validator`.

**Comment by alextrastero** [comment]
Yup, just seen this, I will take a look this evening and should have a fix pronto, thanks for letting me know :)

**Comment by yanickrochon** [comment]
Alright! For the prop types, perhaps have the value as

```
  defaultValue: PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.number
  })
```

?

**Comment by yanickrochon** [comment]
The `Button` component no longer allow setting `onClick` handlers. What's the point of having a button if one cannot handle clicks from it? :stuck_out_tongue:

**Comment by yanickrochon** [comment]
This `Modal` is no longer triggered by it's trigger :

```jsx
<Modal
  header="Create Category"
  trigger={
    <Button node="a" waves="green" flat>New</Button>
  }
  actions={
    <div>
      <Button waves="light" modal="close" className="green" onClick={ this.createCategory.bind(this) }>Create</Button>
      <Button flat modal="close">Dismiss</Button>
    </div>
}>
  <Row>
    <Col s={12} className="input-field">
      <input type="text" ref="newCategory" id="txtNewCategory" />
      <label htmlFor="txtNewCategory">Category Name</label>
    </Col>
  </Row>
</Modal>
```

**Comment by alextrastero** [comment]
Yeah, the modal is broken because of the Button issue. Please let me know more issues that you find here and I will address them all this evening, and thanks again!

**Comment by yanickrochon** [comment]
:+1:

**Comment by alextrastero** [comment]
Hey @yanickrochon, so I have a fix for most of the issues: https://github.com/react-materialize/react-materialize/pull/109
I've passed `name` down to the input and added the proptype validation for `defaultValue`, if you check the code diff you can see that `value` was never accepted as a prop. Since value is a valid input attribute, because of luck, it worked for you, but you should use defaultValue.

**Comment by yanickrochon** [comment]
@alextrastero I am using [controlled inputs](https://facebook.github.io/react/docs/forms.html#advanced-topics) in my forms. If you ignore the `value` props, your are essentially blocking all this feature together.

**Comment by alextrastero** [comment]
Correct again, have a look once more time https://github.com/react-materialize/react-materialize/pull/109

**Comment by yanickrochon** [comment]
:+1:  I'il try this out!

**Comment by alextrastero** [comment]
This should be fixed with #109
